src/controller/genericmidi.cxx

```diff
@@ -609,19 +609,18 @@ void GenericMIDI::launchScene( int scene )
 		Binding* b = actionToMidi.at(i);
 
 		if ( b->action == GRID_LAUNCH_SCENE ) {
-			for( int i = 0; i < 5; i++ ) {
-				if ( i != scene ) {
-					unsigned char data[3];
-					data[0] = b->status;
-					data[1] = b->data + i;
-					data[2] = 0;
-					writeMidi( data );
-				}
-			}
 			unsigned char data[3];
 			data[0] = b->status;
-			data[1] = b->data + scene;
-			data[2] = 127;
+			data[1] = b->data + i;
```

Comment:
There's a bug here I think. The "i" variable is from the actionToMidi vector. In short, the"i" variable can be any value at all - it doesn't relate to the action.

Debug by printing the value of "i" as it is working, and try to identify why it was working as expected - I would expect it to break if you add a button binding at the start of the `.ctlr`output JSON. (Adding an item will change the position of the `scene_launch` binding in the vector, so the `i` variable will have a different value, causing it to have a different effect?

Reply:
This is code i just copied from you, look at current master line 616.

Reply:
When you removed the for() loop, the meaning of the `i` variable changed. In the previous code, i always ranged from 0 to 4, while now it could by anything - depends on how many items the `actionToMidi` vector contains.

The changes proposed here introduce a bug - that needs to be fixed.

Reply:
Oh deer, you are right. Sorry for this, will fix it!